delivery_carrier_state/models/delivery_carrier.py

```diff
@@ -0,0 +1,16 @@
+###############################################################################
+# For copyright and license notices, see __manifest__.py file in root directory
+###############################################################################
+from odoo import fields, models
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = 'delivery.carrier'
+
+    def send_shipping(self, pickings):
+        res = super().send_shipping(pickings)
+        pickings.write({
+            'delivery_state': 'shipping_recorded_in_carrier',
```

Comment:
What if there are pickings not being recorded? Maybe this should be in a separate method and do it only by default for the core methods. What do you think about it?

Reply:
If has not been registered must thrown an exception.

Reply:
Ok :+1: